subprojects/smoke-test/src/smokeTest/groovy/org/gradle/smoketests/AndroidPluginsSmokeTest.groovy

```diff
@@ -113,6 +109,7 @@ class AndroidPluginsSmokeTest extends AbstractSmokeTest {
         result.task(':app:assembleDebug').outcome == (VersionNumber.parse(agpVersion) < VersionNumber.parse("3.6.0")
             ? TaskOutcome.SUCCESS
             : TaskOutcome.UP_TO_DATE)
+        result.task(':app:processDebugAndroidTestManifest').outcome == TaskOutcome.UP_TO_DATE
```

Comment:
❓ is this test supposed to fail if `ImmutablePublishArtifact` doesn't implement `equals`? Because I can't get it to fail.

Reply:
Good catch. There was a dependency missing that triggers the manifest generation for tests. Fixed